controllers/reviewController.js

```diff
@@ -58,6 +59,7 @@ const getSingleReview = async (req, res) => {
 
 const updateReview = async (req, res) => {
   const { id: reviewId } = req.params; //using object destructuring to extract the id property from the params object of the req (request) object and assign it to the reviewId variable.
+  // AKOSREVIEW - what if I only wanted to update one of these things?
   const { rating, title, comment } = req.body; //getting access to rating, title, comment from req,body (from frontend)
```

Comment:
I noticed that if I use the `updateReview` endpoint, and I only send in one field (for example, I only want to update the rating of my review), then it ends up trying to set the other fields to null/undefined. That's because here on this line when we destrucutre, any fields that are not found, Javascript will end up setting to undefined. We should probably check if `req.body` had those fields in the first place. So we could do something like `req.body.hasOwnProperty('rating')`.

Reply:
Looks good! 

You can shorten it a bit by doing the ternary at the same time as when you're assigning the value:

```javascript
const { rating, title, comment } = req.body;
const review = await Review.findOne({ _id: reviewId });

if (!review) {
  throw new CustomError.NotFoundError(`No review with id ${reviewId}`);
}

checkPermissions(req.user, review.user);
review.rating = req.body.hasOwnProperty('rating') ? req.body.rating : review.rating
review.title = req.body.hasOwnProperty('title') ? req.body.title : review.title
review.comment = req.body.hasOwnProperty('comment') ? req.body.rating : review.comment

await review.save();
res.status(StatusCodes.OK).json({ review });
```

Another thought I had, is maybe we don't even need the `hasOwnProperty`. When we do the object destructuring (`const { rating, title, comment } = req.body;`), then any fields that didn't exist in `req.body` will just be saved to the variables on the left-hand-side as `undefined`. So we could do

```javascript
const { rating, title, comment } = req.body;
const review = await Review.findOne({ _id: reviewId });

if (!review) {
  throw new CustomError.NotFoundError(`No review with id ${reviewId}`);
}

checkPermissions(req.user, review.user);

// Assign values from request to the review; fall back to the existing values if the request did not provide such a value
review.rating = rating || review.rating;
review.title = title || review.title;
review.comment = comment || review.comment;

await review.save();
res.status(StatusCodes.OK).json({ review });
```

This says update the `review.rating` field to the `rating` variable OR if that's undefined/null, then to the existing `review.rating` value. 

Another alternative, if you prefer just using clearer `if` statements:

```javascript
const { rating, title, comment } = req.body;
const review = await Review.findOne({ _id: reviewId });

if (!review) {
  throw new CustomError.NotFoundError(`No review with id ${reviewId}`);
}

checkPermissions(req.user, review.user);

// Only assign updated values if they were given in the request:
if (rating) { review.rating = rating }
if (title) { review.title = title }
if (comment) { review.comment = comment }

await review.save();
res.status(StatusCodes.OK).json({ review });
```